apps/internal/base/base.go

```diff
@@ -271,11 +271,8 @@ func (b Client) AuthCodeURL(ctx context.Context, clientID, redirectURI string, s
 }
 
 func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilentParameters) (AuthResult, error) {
-	tenant := silent.TenantID
-	if tenant == "" {
-		tenant = silent.Account.Realm
-	}
-	authParams, err := b.AuthParams.WithTenant(tenant)
+	// when silent.TenantID == "", this is a no-op and we use the client application's configured tenant
```

Comment:
Hmm. If the `tenant == ""`, are you going to use the `tenantId` of the PCA/CCA authority? This should only be the case if that authority's tenant is a GUID or `consumers`, but not `common` or `organizations`. If it's `common` or `organiations` then `Account.HomeTenantId` is the correct tenant to use. 

Note: HomeTenantId is not the same as Realm.

Reply:
Is this correct?
```mermaid
graph TD
    A{caller set WithTenantID?}
    A-->|yes|B
    A-->|no|C
    C[check authority]-->D
    D(GUID or consumers)-->B
    C-->E(common or organizations)
    E-->|PCA|F
    E-->|CCA|G{{error: need authority}}
    B[use that tenant]
    F{have home tenant?}
    F-->|yes|B
    F-->|no|H{{error: need account}}
```

Reply:
I've updated the tenant selection code for silent auth to follow the above flow except for returning an error when a confidential client is configured for the "common" authority. That's the default for confidential clients and although it's incorrect for token requests, silent auth is nonetheless possible with that configuration should the client also be configured to use a custom token provider, because the client won't send a token request in that case.

Reply:
For OBO, the customer should be using `incommingAssertion.tid` and not `common`
For client_credentials, we should never use `common`, that's only for user tokens.
For AcquireTokenSilent, you should always have an `Account` and use `acc.HomeTenantId` in some cases.

But what about `AcquireTokenByAuthorizationCode`? That's part of the confidential client, but it should allow usage of `common`

Reply:
This PR does doesn't change the auth code flow. A confidential client can still have `lmo/common` as its authority and try to authenticate there.

Reply:
> I've updated the tenant selection code for silent auth to follow the above flow **except for** returning an error when a confidential client is configured for the "common" authority

I would recommend updating the diagram, because it will become the much-easier-to-consume medium in the future when someone is reviewing this change.

Other than that, this PR looks good to me. Approving.